### PR TITLE
Cache cabal wrapper to avoid invalidating configuration

### PR DESCRIPTION
**Pull request by wz1000**
Leads to much faster ghcide startup times.

### CODE REVIEW

**Comment by wz1000** [comment]
From CI:   

```
Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
```

**Comment by fendor** [comment]
I can't see where `add-path` is allegedly being used...

**Comment by fendor** [comment]
https://github.com/actions/setup-haskell/issues/44 is related

**Comment by jneira** [comment]
> 
> 
> [actions/setup-haskell#44](https://github.com/actions/setup-haskell/issues/44) is related

It could be avoided with something like https://github.com/haskell/ghcide/pull/912/commits/5b36a5fcd7e75c7ac669acb590b29d33d165dc7d

**Comment by fendor** [comment]
I know, but I am wondering whether it is ok to wait for two days, then update CI once and then merge this...

**Comment by jneira** [comment]
> I know, but I am wondering whether it is ok to wait for two days

Fair enough, not sure if it will be fixed in two days or more though :stuck_out_tongue_closed_eyes:

**Comment by fendor** [comment]
@wz1000 I think you have to rebase on top of master, then CI should run as expected

**Comment by wz1000** [comment]
I'm stumped. @fendor, @jneira help, please?

**Comment by fendor** [comment]
@wz1000 We will get to the bottom of it. I will boot my windows again :) *puts on security helmet*

**Comment by jneira** [comment]
I'll try to take a look too. will share what i find

**Comment by jneira** [comment]
Ok, i've tried locally this version and the command is failing in ci works for me:

* ci
```
HUnitFailure (Just (SrcLoc {srcLocPackage = "main", srcLocModule = "Main", srcLocFile = "tests\\\\BiosTests.hs", srcLocStartLine = 303, srcLocStartCol = 12, srcLocEndLine = 303, srcLocEndCol = 73})) (Reason "Unexpected cradle failError when calling cabal --builddir=C:\\Users\\runneradmin\\AppData\\Local\\hie-bios\\dist-hie-bios-test-0fe1fee0b81853d0-2894c051acc238b93bf25188905c0e36 v2-exec --with-compiler C:\\Users\\runneradmin\\AppData\\Local\\hie-bios\\wrapper-4da94d68375fab32d2147da1837e9611.exe ghc -v0 -- --print-libdir\n\ncabal: The program 'ghc' version >=7.0.1 is required but the version of\nC:\\Users\\runneradmin\\AppData\\Local\\hie-bios\\wrapper-4da94d68375fab32d2147da1837e9611.exe\ncould not be determined.\n\n\n")
```
* locally

```
PS D:\dev\ws\haskell\hie-bios> cabal --builddir=C:\Users\atrey\AppData\Local\hie-bios\dist-hie-bios-85a445ee2bd89b92376799d3a4db4f95  v2-exec --with-compiler C:\Users\atrey\AppData\Local\hie-bios\wrapper-4da94d68375fab32d2147da1837e9611.exe ghc -v0 -- --print-libdir
D:\bin\stack\x86_64-windows\ghc-8.10.1\lib
```

It seems the wrapper in ci does not have ghc con `$PATH` (but it must be!)

File: src/HIE/Bios/Cradle.hs [hunk 1]
```diff
@@ -406,8 +406,9 @@ cabalCradle wdir mc =
             -- ./dist-newstyle/tmp/environment.-24811: createDirectory: does not exist (No such file or directory)
             createDirectoryIfMissing True (buildDir </> "tmp")
             -- Need to pass -v0 otherwise we get "resolving dependencies..."
+            wrapper_fp <- withCabalWrapperTool ("ghc", []) wdir
             readProcessWithCwd
-              wdir "cabal" (["--builddir="<>buildDir,"v2-exec", "ghc", "-v0", "--"] ++ args) ""
+              wdir "cabal" (["--builddir="<>buildDir,"v2-exec","--with-compiler", wrapper_fp, "ghc", "-v0", "--"] ++ args) ""
```
Comment (jneira) [inline src/HIE/Bios/Cradle.hs:411]:
```suggestion
              wdir "cabal" (["v2-exec", "ghc", "-v0", "--"] ++ args) ""
```

what about if we use the hie-bios cabaldir for cabal repl, but the user cabaldir for cabal exec (without -w)? that way cabal exec will not invalidate  neither

**Pull request merged**